src/thunderbots/software/network_input/networking/network_client.cpp

```diff
@@ -90,12 +90,20 @@ void NetworkClient::filterAndPublishVisionData(SSL_WrapperPacket packet)
             ret.first->second = detection;
         }
 
+        std::optional<Timestamp> latest_timestamp = std::nullopt;
         // Create a vector of the latest detection data for each camera
         std::vector<SSL_DetectionFrame> latest_detections;
         for (auto it = latest_detection_data.begin(); it != latest_detection_data.end();
              it++)
         {
             latest_detections.push_back(it->second);
+            Timestamp timestamp = Timestamp::fromSeconds(it->second.t_capture());
+
+            // Remember the latest timestamp
+            if (!latest_timestamp || timestamp > latest_timestamp)
```

Comment:
Nit:
I like seeing "!latest_timestamp.has_value()", I feel like it's easier to understand the datatype at a glance and what you're checking.
Up to you though.